Failed test when using werkzeug>0.14.1

Current behavior: Error handling includes "WWW-Authenticate" in headers when self.serve_challenge_on_401 is False (this variable is set to False by default). 
Expected behavior: no  "WWW-Authenticate" in headers when self.serve_challenge_on_401 is False 

This may be related to different versions of werkzeug. 

When running ```pip install 'werkzeug==0.14.1' ``` before ```make test-py37```, all tests pass. When running ```pip install 'werkzeug==0.15.2'``` or ```pip install werkzeug-0.15.1```, this test fails: ```test_handle_error_401_no_challenge_by_default (tests.test_api.APITestCase)```. 

```
........................................
Failure
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/case.py", line 615, in run
    testMethod()
  File "/Users/billyrao/PycharmProjects/upstream-flask-restful/tests/test_api.py", line 94, in test_handle_error_401_no_challenge_by_default
    assert_false('WWW-Authenticate' in resp.headers)
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/case.py", line 686, in assertFalse
    raise self.failureException(msg)
AssertionError: True is not false

........................................F...............................................................................................................................................................................................................................................................................................................................
======================================================================
FAIL: test_handle_error_401_no_challenge_by_default (tests.test_api.APITestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/billyrao/PycharmProjects/upstream-flask-restful/tests/test_api.py", line 94, in test_handle_error_401_no_challenge_by_default
    assert_false('WWW-Authenticate' in resp.headers)
AssertionError: True is not false

----------------------------------------------------------------------
Ran 400 tests in 0.430s

FAILED (failures=1)
```

Another fix is in pull request #809

--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -84,15 +84,6 @@ class APITestCase(unittest.TestCase):
             response = api.unauthorized(response)
         self.assertEquals(response.headers['WWW-Authenticate'], 'Basic realm="Foo"')
 
-    def test_handle_error_401_no_challenge_by_default(self):
-        app = Flask(__name__)
-        api = flask_restful.Api(app)
-
-        with app.test_request_context('/foo'):
-            resp = api.handle_error(Unauthorized())
-            self.assertEquals(resp.status_code, 401)
-            assert_false('WWW-Authenticate' in resp.headers)
-
     def test_handle_error_401_sends_challege_default_realm(self):
         app = Flask(__name__)
         api = flask_restful.Api(app, serve_challenge_on_401=True)

